### DISCUSSION TITLE
How not to pass fetch to every lib function

### DISCUSSION THREAD

**Discussion by wvhulle** [sveltejs/kit#9033] [General]
I have a large app with a lot of lib code. The page endpoints call this lib code from different points in my app. Because load is only available in load functions, I have to pass fetch to every single function that calls my backend for getting or posting some json. 

I find it very ugly. I have tried assigning fetch to a global variable but that didn't work. In the source code of kit I cannot find an export for the load function. How can I import it (at most once per lib file) and drop all those fetch arguments?

**Comment by schwartzmj** [comment]
I have a similar use-case and am also interested in solutions.

**Comment by pepa-linha** [comment]
I solve this problem by having an (ES6) module for the client, for which I set fetch from SvelteKit during the life cycle. Then I import this module in the places where I need it and all API calls go from there.

**Reply by schwartzmj** [reply]
@pepa-linha Apologies, I'm not really following what you're describing. If you have a moment, would you be able to elaborate maybe with an example?

**Reply by david-plugge** [reply]
If I understand this correct... You are doing something dangerous! It sounds like you are using a global variable to store `fetch` from a load function and use that in your api. That is very very bad because two different requests can and will use the same fetch function. That means that multiple requests for **different** users will have the same cookies as sveltekit does some internal stuff.
Im sorry if it's hard to follow along, just don't ever store anything that's coming in as a load function parameter inside global variables unless you are 100% sure of what you are doing!

**Comment by Gustutu** [comment]
i have the same problem. Looking for a solution :) .

**Comment by david-plugge** [comment]
This is a known limitation and since we are approaching version 2 it is time to bring it up again. A new issue should be created or if there is an open one already make some noice over there.
